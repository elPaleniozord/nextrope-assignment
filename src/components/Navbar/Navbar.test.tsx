import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router';
import { store } from '../../app/store';
import Menu from './Navbar';

test('should render navbar component', () => {
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Menu />
      </MemoryRouter>
    </Provider>
  )
  expect(getByText(/Logo/)).toBeInTheDocument()
  expect(getByText(/Książki/)).toBeInTheDocument()
  expect(getByText(/Koszyk/)).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})

// test('should handle navigation', () => {
//   let mockHistory, mockLocation
//   const { getByText } = render(
//     <Provider store={store}>
//       <MemoryRouter initialEntries={["/my-initial-route"]}>
//         <Menu />
//         <Routes >
//           <Route path="*" render={({history, location}) => {
//             mockHistory = history
//             mockLocation = location
//             return null
//           }} />
//         </Routes>
//       </MemoryRouter>
//     </Provider>
//   )
//   userEvent.click(getByText(/Koszyk/))
//   console.log(getByText(/Koszyk/))
//   expect(mockLocation.pathname).toBe("/cart")
// })