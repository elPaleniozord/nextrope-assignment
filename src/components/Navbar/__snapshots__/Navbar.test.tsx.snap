// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render navbar component 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-primary"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Logo
      </a>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Książki
        </a>
        <a
          class="nav-link"
          href="/cart"
        >
          Koszyk 
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
